quinn-proto/src/tests/mod.rs

```diff
@@ -1177,6 +1177,39 @@ fn keep_alive() {
     }
 }
 
+#[test]
+fn cid_retirement() {
+    let _guard = subscribe();
+    let mut pair = Pair::default();
+    let (client_ch, server_ch) = pair.connect();
+
+    // Server retires current active remote CIDs
+    pair.server_conn_mut(server_ch).mock_retire_cid(1, 1);
+    pair.drive();
+    // Any unexpected behavior may trigger TransportError::CONNECTION_ID_LIMIT_ERROR
+    assert!(!pair.client_conn_mut(client_ch).is_closed());
+    assert!(!pair.server_conn_mut(server_ch).is_closed());
+    assert_matches!(pair.client_conn_mut(client_ch).active_rem_cid_seq(), 1);
```

Comment:
Client holds active remote cids 0,1,2,3,4. When server actively retire cid 0, current version will respond with CONNECTION_ID_LIMIT_ERROR which is not true. The reason is because current code run `update_rem_cid ` after `insert`.